src/Microsoft.IdentityModel.Xml/XmlTokenStream.cs

```diff
@@ -116,8 +119,10 @@ public void WriteTo(XmlWriter writer)
             if (writer == null)
                 throw LogArgumentNullException(nameof(writer));
 
-            var streamWriter = new XmlTokenStreamWriter(_entries, _excludedElement, _excludedElementNamespace);
-            streamWriter.WriteTo(writer);
+            var streamWriter = new XmlTokenStreamWriter(this);
+            streamWriter.WriteTo(writer, _excludedElement, _excludedElementNamespace);
         }
+
+        internal ReadOnlyCollection<XmlToken> XmlTokens => _xmlTokens.AsReadOnly();
```

Comment:
Can we return return non-readonly collection here? Perhaps it is perfectly valid to leave it as readonly by design, but I wanted to clarify.
Thinking from the perspective of EnvelopedSignatureWriter, the only way to manipulate XmlToken position or content of an XmlToken within the stream is to create a new stream out of the existing one.

Reply:
I think we have to leave it readonly, otherwise we could get different results.